REST_API_Development/WebApiPurpose.md

```diff
@@ -0,0 +1,39 @@
+# Web-API purpose
+
+## What is a [Web API](https://en.wikipedia.org/wiki/Web_API)?
+
+A Web API is an application programming interface located on a web-server. It manages the retrieval of requests as well as passing along an associated responses through the HTTP/HTTPS protocol. Such requests are usually in the format of JavaScript Object Notation (Json) or Extensible Markup Language (XML).
+
+## Why the repository exists and whats its used for?
+
+The web-API repository is the backend of the [Giraf Project](https://github.com/aau-giraf/web-api), and it is built with the .NET Core in C#. More specifically the Web API is responsible for the handling of all communication coming from and to the API-Client as shown Figure 7.16. If it is a valid request this leads to an [endpoint](./EndpointsAndControllers.md) and if not then an error response is returned. Most request would require the web-API to communicate with the [database](./Database.md) that stores all the data related to the Giraf Project, which allows several of components to be communicating information to the common database.
```

Comment:
"and it is built with the .NET Core in C#", instead: "and it is programmed in C# and uses the .NET Core framework."